adding opensearch as embedding store (#213)
quarkiverse/quarkus-langchain4j: docs/modules/ROOT/pages/opensearch-store.adoc

= OpenSearch Document Store for Retrieval Augmented Generation (RAG)

include::./includes/attributes.adoc[]

> When implementing Retrieval Augmented Generation (RAG), a capable document store is necessary. This guide will explain how to leverage a pgvector database as the document store.

Reviewer: ```suggestion
When implementing Retrieval Augmented Generation (RAG), a capable document store is necessary. This guide will explain how to leverage an _opensearch-store_ database as the document store.
```

Maybe add a link to opensearch.

== Leveraging the OpenSearch Document Store

To utilize the OpenSearch document store, you'll need to include the following dependency:

[source,xml,subs=attributes+]
----
<dependency>
    <groupId>io.quarkiverse.langchain4j</groupId>
    <artifactId>quarkus-langchain4j-opensearch</artifactId>
    <version>{project-version}</version>
</dependency>
----

This extension relies on the OpenSearch Java Client, make sure you have one configured correctly.

> Upon installing the extension, you can utilize the pgvector store using the following code:

Reviewer: ```suggestion
Upon installing the extension, you can utilize the opensearch store using the following code:
```

[source,java]
----
include::{examples-dir}/io/quarkiverse/langchain4j/samples/IngestorExampleWithOpenSearch.java[]
----

== Configuration Settings

Customize the behavior of the extension by exploring various configuration options:

> include::includes/quarkus-langchain4j-pgvector.adoc[leveloffset=+1,opts=optional]

Reviewer: ```suggestion
include::includes/quarkus-langchain4j-opensearch.adoc[leveloffset=+1,opts=optional]
```

== Under the Hood

> Each ingested document is saved as a row in a Postgres table, containing the embedding column stored as a vector.

Reviewer: That's need to be updated.
